src/adapters/socket.io/index.ts

```diff
@@ -18,10 +18,11 @@ class SocketIOAdapter extends Adapter {
   }
 
   _connect(): Promise<this> {
-    return new Promise((resolve) => {
+    return new Promise(async (resolve) => {
```

Comment:
```suggestion
    return new Promise( (resolve) => {
```

I think promise executor function should not be `async`, how do we call an async function in a promise executor function?